pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -331,6 +331,17 @@ type PipelineRunStatusFields struct {
 
 	// PipelineRunSpec contains the exact spec used to instantiate the run
 	PipelineSpec *PipelineSpec `json:"pipelineSpec,omitempty"`
+
+	// list of tasks that were skipped due to when expressions evaluating to false
+	// +optional
+	SkippedTasks []SkippedTask `json:"skippedTasks,omitempty"`
+}
+
+// SkippedTask is used to describe the Tasks that were skipped due to their When Expressions
+// evaluating to False
+type SkippedTask struct {
```

Comment:
nice 👍

Reply:
might be worth a comment explaining why this is a struct - or part of the commit message? nbd tho